src/language/from_relay/mod.rs

```diff
@@ -655,29 +671,61 @@ fn compile_expression(
         .clone()
         .downcast::<tvm::ir::relay::TupleGetItem>()
     {
-        // We've only implemented tuple functionality for handling batch norms,
-        // at the moment. So we check to make sure this is handling a batch
-        // norm.
+        // For batch norms, we use a hack where if
+        // simplify_batch_norm_for_inference_hack is set,
+        // we compile the Relay batch norm to a single output.
+
+        // All other expressions are compiled using Language::TupleGetItem construct
 
         let arg = tuple_get_item
             .tuple
             .clone()
-            .downcast::<tvm::ir::relay::Call>()
-            .unwrap();
-        assert_eq!(
-            arg.op
-                .clone()
-                .upcast::<tvm::ir::expr::BaseExpr>()
-                .downcast::<tvm::ir::op::Op>()
-                .unwrap()
-                .name
-                .as_str(),
-            Ok("nn.batch_norm")
-        );
-        assert_eq!(tuple_get_item.index, 0);
+            .downcast::<tvm::ir::relay::Call>();
```

Comment:
The `tuple` field might not be a `Call` anymore! For example, if we have something like `(<something>, <something else>).1`, then the `tuple` field of the `tuple_get_item` will be a tuple constructor.

This can be fixed *and* the logic of this `else if` branch can be simplified a lot, all at once. I think the following structure makes sense:
First, check if `simplify_batch_norm_for_inference_hack` is true && if the `tuple` is a batch norm && if `tuple_get_item.index == 0`. If all three of those things are true, then we should apply the hack. The hack, in this case, is the fact that we don't create a `TupleGetItem`, but instead, we directly return the batch norm (`get_compiled_expression(tuple_get_item.tuple.clone())`).
If the hack doesn't fire, then just generate the `TupleGetItem` as you already do on lines 707-710.

Can you check my logic on that? I think it makes sense, but it might not.

Reply:
yep that makes sense! im realizing now that we may run into a little trouble if `tuple_get_item.tuple` isn't a `Op`.

Also, I have duplicated code here to comply with Rust's type checking. If there's any simpler way to do this lmk!

Reply:
Simplified this code a bit, we're good here now!